parser/proto.y

```diff
@@ -327,269 +334,231 @@ stringLit : _STRING_LIT {
-		} else {
-			$$ = $2
-		}
-	}
-	| {
-		$$ = nil
+messageTextFormat : messageLiteralFields {
+		$$ = $1
 	}
 
-aggField : aggFieldEntry {
+messageLiteralFields : messageLiteralField ',' messageLiteralFields {
 		if $1 != nil {
-			$$ = &messageFieldEntry{$1, nil}
+			entry := &messageFieldEntry{$1, $2}
+			$$ = &messageFieldList{entry, $3}
 		} else {
 			$$ = nil
 		}
 	}
-	| aggFieldEntry ',' {
+	| messageLiteralField ';' messageLiteralFields {
 		if $1 != nil {
-			$$ = &messageFieldEntry{$1, $2}
-		} else {
-			$$ = nil
-		}
+			entry := &messageFieldEntry{$1, $2}
+                	$$ = &messageFieldList{entry, $3}
+                } else {
+                	$$ = nil
+       		}
 	}
-	| aggFieldEntry ';' {
+	| messageLiteralField messageLiteralFields {
 		if $1 != nil {
-			$$ = &messageFieldEntry{$1, $2}
+			entry := &messageFieldEntry{$1, nil}
+                	$$ = &messageFieldList{entry, $2}
 		} else {
 			$$ = nil
 		}
 	}
-	| error ',' {
-		$$ = nil
-	}
-	| error ';' {
-		$$ = nil
-	}
-	| error {
-		$$ = nil
-	}
-
-aggFieldEntry : aggName ':' scalarConstant {
+	| messageLiteralField ',' {
 		if $1 != nil {
-			$$ = ast.NewMessageFieldNode($1, $2, $3)
+			entry := &messageFieldEntry{$1, $2}
+                	$$ = &messageFieldList{entry, nil}
 		} else {
 			$$ = nil
 		}
 	}
-	| aggName '[' ']' {
+	| messageLiteralField ';' {
 		if $1 != nil {
-			val := ast.NewArrayLiteralNode($2, nil, nil, $3)
-			$$ = ast.NewMessageFieldNode($1, nil, val)
+			entry := &messageFieldEntry{$1, $2}
+			$$ = &messageFieldList{entry, nil}
 		} else {
 			$$ = nil
 		}
 	}
-	| aggName ':' '[' ']' {
+	| messageLiteralField {
 		if $1 != nil {
-			val := ast.NewArrayLiteralNode($3, nil, nil, $4)
-			$$ = ast.NewMessageFieldNode($1, $2, val)
+			entry := &messageFieldEntry{$1, nil}
+			$$ = &messageFieldList{entry, nil}
 		} else {
 			$$ = nil
 		}
 	}
-	| aggName '[' msgList ']' {
-		if $1 != nil {
-			vals, commas := $3.toNodes()
-			val := ast.NewArrayLiteralNode($2, vals, commas, $4)
-			$$ = ast.NewMessageFieldNode($1, nil, val)
-		} else {
-			$$ = nil
-		}
-	}
-	| aggName ':' '[' constantList ']' {
-		if $1 != nil {
-			vals, commas := $4.toNodes()
-			val := ast.NewArrayLiteralNode($3, vals, commas, $5)
-			$$ = ast.NewMessageFieldNode($1, $2, val)
-		} else {
-			$$ = nil
-		}
-	}
-	| aggName ':' '[' error ']' {
-		$$ = nil
-	}
-	| aggName '[' error ']' {
+	| {
 		$$ = nil
 	}
-	| aggName ':' msgLit {
-		if $1 != nil && $3 != nil {
+
+messageLiteralField : messageLiteralFieldName ':' value {
+		if $1 != nil {
 			$$ = ast.NewMessageFieldNode($1, $2, $3)
 		} else {
 			$$ = nil
 		}
 	}
-	| aggName msgLit {
-		if $1 != nil && $2 != nil {
+	| messageLiteralFieldName messageValue {
+		if $1 != nil {
 			$$ = ast.NewMessageFieldNode($1, nil, $2)
 		} else {
 			$$ = nil
 		}
 	}
-	| aggName ':' '<' error '>' {
+	| error ',' {
+		$$ = nil
+	}
+	| error ';' {
 		$$ = nil
 	}
-	| aggName '<' error '>' {
+	| error {
 		$$ = nil
 	}
```

Comment:
These seem out of place and are confusing here. They refer to trailing `,` or `;` punctuation, but those are not otherwise seen in any of the non-error productions above. I think this is fallout from changing the structure and not keeping an analog to `aggFieldEntry` (so along same lines as comment above).

Reply:
Now we are using `messageLiteralFieldEntry` production, so the error handling is happening in `messageLiteralFieldEntry`.